kubernetes/kubernetes.go

```diff
@@ -370,6 +370,13 @@ func (in *K8SClient) GetDeployment(namespace, name string) (*apps_v1.Deployment,
 // GetDeployments returns an array of deployments for a given namespace.
 // It returns an error on any problem.
 func (in *K8SClient) GetDeployments(namespace string, opts meta_v1.ListOptions) ([]apps_v1.Deployment, error) {
+	if namespace != "" {
+		if depList, err := in.k8s.AppsV1().Deployments(namespace).List(in.ctx, opts); err == nil {
+			return depList.Items, nil
+		} else {
+			return []apps_v1.Deployment{}, err
+		}
+	}
```

Comment:
What am I missing here? Why is this code inside the outer if-stmt identical code to the code below? If namespace != "" then do this code... (which then returns - the inner if stmt will return in the if or else - you never drop out).

But then if namespace == "" then drop down and ... run the same exact code???

I don't know why this code was added - I think you can delete 373-379 (the code that was added) unless there's something different that I can't see.

Reply:
Yeah, that looks like an unfinished edit. I'll look at it.

Reply:
I think the change here was to guard  making the call to Deployments for an empty string. Perhaps something changed in the calling patterns and I found it necessary. I think the guard is fine/necessary.  But, despite already fixing Mazz's issue, the way it's written is still verbose. I'll update it.

Reply:
I changed my mind, I don't think we should block namespace = "". I think the convention there is that `namespace == ""`  means all namespaces.